### DISCUSSION TITLE
issues with booting (only) some LIVE linux ISOs

### DISCUSSION THREAD

**Discussion by wolfie-cz** [ipxe/ipxe#1041] [General]
Hi, 
I would kindly as for help with booting some of the livelinux ISOs over http / internet
My goal is to set a "light copy" of netboot.xyz  (only some distros) on my home network so that I was able to load/boot live isos faster than over internet from netboot.xyz. And also to be able boot legally purchased SW (Lazesoft,MiniBootPartition) over the LAN and also internet.

I am having troubles with booting all ubuntu and debian distros, while some linux forks like wubuntu (windowsFX), Zorin or ElementaryOS boot normally.

As an example : Ubuntu 23.04-Lxqt-LIVE
So that I was able to set the boot, I downloaded the ipxe file from _http://boot.netboot.xyz/live-ubuntu.ipxe_ and I chose line 21 _(item ubuntu-23.04-LXQT-squash ${space} Ubuntu 23.04 Lxqt)_
the following lines are relevant: (lines 154, 155,156,180,181,182,183):
```
:ubuntu-23.04-LXQT-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/23.04-5eee26d9/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/23.04-5eee26d9/1
imgfree
kernel ${kernel_url}vmlinuz ip=dhcp boot=casper netboot=http fetch=${squash_url} initrd=initrd.magic ${cmdline}
initrd ${kernel_url}initrd
boot

```
So I downloaded the files are needed for netboot:
```
https://github.com/netbootxyz/ubuntu-squash/releases/download/23.04-5eee26d9/filesystem.squashfs
https://github.com/netbootxyz/ubuntu-squash/releases/download/23.04-5eee26d9/filesystem.squashfs.part2
https://github.com/netbootxyz/ubuntu-squash/releases/download/23.04-5eee26d9/vmlinuz
https://github.com/netbootxyz/ubuntu-squash/releases/download/23.04-5eee26d9/initrd

```
I took these files and saved them in the directory and added a record in my menu.ipxe accordingly.

```
:ubuntu-23.04-LXQT-squash
set cmdline console=ttyS0,115200n8
set squash_url ${my_public_IP}/ubuntu2304/filesystem.squashfs
set kernel_url ${my_public_IP}/ubuntu2304/
imgfree
kernel ${kernel_url}vmlinuz ip=dhcp boot=casper netboot=http fetch=${squash_url} initrd=initrd.magic ${cmdline}
initrd ${kernel_url}initrd
boot

```

But when I try to boot the system, althought all three files load fine (I can see the progress of loading and then OK at the end), there is a frozen black screen, no matter if I try booting virtualbox or real PC, LAN or internet...

When this did not work I tried many other distros from:
https://github.com/netbootxyz/ubuntu-squash/releases/
and
https://github.com/netbootxyz/debian-squash/releases

but the result is still the same - first the files are loading, I can see OK the the end of the lines, then a the following message appeares shortly: EFI stub : Loaded initrd from **LINUX_EFI_INITRD_MEDIA_GUID device path** and then only black screen.
First I thought that the EFI Stub message was the issue, but later I realized that this message appears also while booting ElementaryOS od WindowsFX, but after that message these (Elementary and WIndowsFX) boot notmally.

I **was unable** to boot over LAN/internet these distros: Ubuntu 23.04,22.04.20.04,18.04, Debian 12,11,10, Kali

While I **was able to manage** boot the following (using the same way- copy the pattern from netboot.xyz and change the path):
Zorin 16.1, 16.3 Lite and Core, Bodhi6, Fatdog-Puppy, ElementaryOS 6 and 7, PartedMagic 2020, WindowsFX 11.4 (wubuntu).

Also, all Windows PE images can boot without any issue (Lazesoft Recovery, Minitool Partition, Bob Ombs)

I just cannot understand why when booting the file same files from netboot.xyz it is boots fine but not from my server and why some distros I can boot and some not 

I also tried downloading live ISOs from the propietary pages for ex. https://www.kali.org/get-kali/#kali-live,
also tried using files from https://http.kali.org/kali/dists/kali-rolling/main/installer-amd64/current/images/netboot/netboot.tar.gz as described here https://www.kali.org/docs/installation/network-pxe/

With my limited knowlegde I am not able to crack this one without some help.

Thank you for any thoughts or hints!
[menu.ipxe.txt](https://github.com/ipxe/ipxe/files/12657005/menu.ipxe.txt)

**Comment by NiKiZe** [comment]
You have console=try... That should give "a black screen"

You can search this repo on GitHub for info related to the LINUX_EFI_INITRD_MEDIA_GUID message, some older kernel have issues. 

Check which version of iPXE you are using, it is the latest git master, or are you using a binary from somewhere else in the working scenario compared to the non working?

I would debug this "issue" by just booting the kernel, when that "works" add the initrd and so forth.
Also your ${my_public_IP} is not defined, does it start with http:// so that the squash fetch which runs inside the os in init scripts and is different between distros works correctly?

**Comment by samysousa** [comment]
I got the same result , black screen after loading. 

`kernel ${kernel_url}vmlinuz ip=dhcp boot=casper netboot=http fetch=${squash_url} initrd=initrd.magic ${cmdline}`   == Blackscreen on some distros

change to :
`kernel ${kernel_url}vmlinuz ip=dhcp boot=casper netboot=url url=${squash_url} initrd=initrd.magic ${cmdline} `  == Works with 23.04 OR (will say wget error and never tries downloading the squashfs) 

You have to experiment between the two or try gramming vmlinuz and initrd files from other distros (That's what i did to find that the 23.04 files work)
